lib/cadet/assessments/assessments.ex

```diff
@@ -517,26 +499,29 @@ defmodule Cadet.Assessments do
         grader = %User{role: role}
       )
       when is_ecto_id(submission_id) and is_ecto_id(question_id) do
-    if role in [@grading_role, @see_all_submissions_role] do
+    if role in (@grading_roles ++ @see_all_submissions_roles) do
       answer_query =
         Answer
         |> where(submission_id: ^submission_id)
         |> where(question_id: ^question_id)
 
       answer =
-        role
-        |> case do
-          @grading_role ->
+        cond do
+          role in @grading_roles ->
             students = Cadet.Accounts.Query.students_of(grader)
 
             answer_query
             |> join(:inner, [a], s in assoc(a, :submission))
             |> join(:inner, [a, s], t in subquery(students), t.id == s.student_id)
+            |> Repo.one()
 
-          @see_all_submissions_role ->
+          role in @see_all_submissions_roles ->
             answer_query
+            |> Repo.one()
+
+          true ->
```

Comment:
This is not even possible since we have made sure that `role in (@grading_roles ++ @see_all_submissions_roles)`, so remove this part, then move the `Repo.one()` to outside the cond.

Anyway, this should've been a `nil` as we're doing `Repo.one()` and not `Repo.all()`

Reply:
Yup was trying to find a clean way of implementing this. Improved!